### ISSUE TITLE
Suggest getting the span of List<T> for iterations

### ISSUE THREAD

**Issue by Meir017** [dotnet/roslyn-analyzers#7344]
### Describe the problem you are trying to solve

consider the following code:
```cs
public void Method(List<int> list)
{
    // should trigger the analyzer
    foreach (var item in list)
    {
        Console.WriteLine(item);
    }
    list.ForEach(Console.WriteLine);

    // the suggested fix
    foreach (var item in CollectionsMarshal.AsSpan(list))
    {
        Console.WriteLine(item);
    }
}
```

Iterating over List<T> without getting the span first.

### Describe suggestions on how to achieve the rule

- check for invocation of the `ForEach` method on when the target is `List<T>`
- check for foreach loops where the collection is `List<T>`

**Comment by sharwell** [comment]
💭 I'm not sure the gains for this are worth giving up the `_version` check.
